### PR TITLE
feat: add siop-oid4vp package

### PR DESCRIPTION
**Pull request by auer-martin**
This PR moves the content from the siop-oid4vp repository from https://github.com/Sphereon-Opensource/SIOP-OID4VP, to this monorepo (siop-oid4vp package). See the discussion [here](https://github.com/Sphereon-Opensource/SIOP-OID4VP/pull/82) for more context

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
     "fix:prettier": "prettier --write \"{packages,__tests__,!dist}/**/*.{ts,tsx,js,json,md,yml}\"",
     "build": "pnpm -r --stream build",
     "build:clean": "lerna clean -y && pnpm install && lerna run build:clean --concurrency 1",
-    "test:ci": "jest --config=jest.json",
-    "test": "jest --verbose --config=jest.json --coverage=true --detectOpenHandles",
+    "test:ci": "jest --config=jest.json && jest --config=packages/siop-oid4vp/jest.json",
```
Comment (TimoGlastra) [inline package.json:15]: Can't we combine this into one jest run?

File: packages/siop-oid4vp/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
+# Release Notes
+
+The DID Auth SIOP typescript library is still in an beta state at this point. Please note that the interfaces might
+still change a bit as the software still is in active development.
+
+## 0.6.5
```
Comment (TimoGlastra) [inline packages/siop-oid4vp/CHANGELOG.md:6]: @nklomp how do you want to deal with versioning? Do we want to adjust it based on the current oid4vci version and then release them at the same time?

File: packages/siop-oid4vp/LICENSE
Comment (TimoGlastra) [inline packages/siop-oid4vp/LICENSE:1]: Can be removed i think, with monorepo license

File: packages/siop-oid4vp/README.md [hunk 3]
```diff
@@ -0,0 +1,1078 @@
...
+<h1 align="center">
+  <br>
+  <a href="https://www.sphereon.com"><img src="https://sphereon.com/content/themes/sphereon/assets/img/logo.svg" alt="Sphereon" width="400"></a>
+  <br><center>Self Issued OpenID Provider (SIOPv2)<br/>
+with OpenID4VP support</center>
+  <br>
+</h1>
+<br>
+
+[![CI](https://github.com/Sphereon-Opensource/SIOP-OpenID4VP/actions/workflows/main.yml/badge.svg)](https://github.com/Sphereon-Opensource/SIOP-OpenID4VP/actions/workflows/main.yml) [![codecov](https://codecov.io/gh/Sphereon-Opensource/SIOP-OpenID4VP/branch/develop/graph/badge.svg?token=9P1JGUYA35)](https://codecov.io/gh/Sphereon-Opensource/SIOP-OpenID4VP) [![NPM Version](https://img.shields.io/npm/v/@sphereon/did-auth-siop.svg)](https://npm.im/@sphereon/did-auth-siop)
```
Comment (TimoGlastra) [inline packages/siop-oid4vp/README.md:11]: The urls are outdated

File: packages/siop-oid4vp/LICENSE
Comment (nklomp) [reply packages/siop-oid4vp/LICENSE:1]: TBH I rather keep them in the individual module folders, like we do everywhere

File: package.json (hunk 1 above)
Comment (auer-martin) [reply package.json:15]: The siop/vp repo had different ts features turned on.
With the current tsconfig of the openid4vc repo, there are many errors.
That's why I created a separate jest.json file. Its only purpose is to use a different tsconfig.
I quickly checked if it was possible to configure it in one file but couldn't get it working.
I think the correct approach is to refactor the repo to align with the stricter typing used elsewhere in the repo at some point.

Comment (TimoGlastra) [reply package.json:15]: Okay that makes sense, and probably good to first get it merged before changing too much about the code

File: packages/siop-oid4vp/README.md (hunk 3 above)
Comment (auer-martin) [reply packages/siop-oid4vp/README.md]: Removed them now. As this information is available in the top level readme

File: package.json (hunk 1 above)
Comment (nklomp) [reply package.json:15]: Yeah, I noticed 2 months ago we had been developing the SIOPv2 package with strict mode off etc. Will fix that soon as a first step before going into the bigger refactor

**Review by nklomp — APPROVED** [review]

**Comment by nklomp** [comment]
LGTM. Thanks for all the work

**Pull request merged**